### DISCUSSION TITLE
爱快作为主路由，设置静态路由至 ppgw 之后， Clash 日志里的来源 IP 全部为爱快 IP

### DISCUSSION THREAD

**Discussion by imKnect** [kkkgo/PaoPaoGateWay#26] [问和答] [answered]
我的 iKuai 地址为 10.1.1.1，ppgw 在 10.1.1.2，ppdns 在 10.1.1.5，配置完毕后可以科学上网，但是除了 socks5 链接进来的可以正常显示来源 IP，其余内网设备科学上网全部显示来源 IP 为 10.1.1.1。

![image](https://github.com/kkkgo/PaoPaoGateWay/assets/36035854/3862d8b1-95ec-4b69-9325-860ac8722533)

![image](https://github.com/kkkgo/PaoPaoGateWay/assets/36035854/7b783ef4-b8a9-475f-bef4-45d56877611d)

怀疑是爱快实现的问题，期望有人能解惑甚至解决……

**Comment by kkkgo** [comment]
也许这个可以说明原因？
https://www.ikuai8.com/index.php?option=com_content&view=article&id=75:nat&catid=21&Itemid=149

**Reply by kkkgo** [reply]
https://bbs.ikuai8.com/forum.php?mod=viewthread&tid=140219&highlight=%E9%9D%99%E6%80%81%E8%B7%AF%E7%94%B1

**Reply by kkkgo** [reply]
https://bbs.ikuai8.com/forum.php?mod=viewthread&tid=87052&highlight=%E9%9D%99%E6%80%81%E8%B7%AF%E7%94%B1%2B%E6%BA%90

**Reply by kkkgo** [reply]
根据爱快论坛描述，需要添加相关过滤规则，我也没怎么深度用过爱快，你可以探索研究下再反馈。

**Comment by kkkgo** [comment]
可以加个`dhcp option`优化下，参考 https://github.com/kkkgo/PaoPaoGateWay/discussions/25#discussioncomment-7221895

**Comment by imKnect** [comment, marked as answer]
![image](https://github.com/kkkgo/PaoPaoGateWay/assets/36035854/0dda5536-440c-43fa-b151-e6e7555986d1)

根据开发者 @kkkgo 给出的上述信息，在 NAT 规则内添加一条目的地为 FakeIP Range 的过滤规则，即可正常显示来源 IP。
原因应该是爱快认为访问的 FakeIP 是外网地址，所以进行了 NAT。

感谢 @kkkgo  的帮助！

**Reply by TwoBall** [reply]
如果设置了TG、网飞的静态路由，也可以一并加上。

**Comment by Makerlong** [comment]
老哥请教一下，爱快的话应该是咋配置的，因为刚用ikuai不是很熟秀，麻烦老哥指点一下

**Comment by TwoBall** [comment]
今天折腾时候发现个问题，爱快如果不使用【静态路由】而使用【下一跳网关】配置fakeip到ppgw，再添加NAT过滤的话，访问就会出现问题。有老哥研究下原因吗？

**Reply by hackerlengyue** [reply]
我已经抛弃paopao了，目前是ikuai+mac mini m2 + surge ，paodns连接数过高会经常重播很烦。

**Reply by TwoBall** [reply]
是看抖音连接数高吗？我的会表现为光猫重启。我一直以为是我的vsol v2802rh固件问题或者是被运营商限制了

**Reply by kkkgo** [reply]
参考【PaoPaoDNS使用技巧】自定义递归对接服务器”外包“ 
 https://github.com/kkkgo/PaoPaoDNS/discussions/131

**Reply by suhx328** [reply]
> 今天折腾时候发现个问题，爱快如果不使用【静态路由】而使用【下一跳网关】配置fakeip到ppgw，再添加NAT过滤的话，访问就会出现问题。有老哥研究下原因吗？

兄弟研究出来了吗，我也遇到了

**Comment by Makerlong** [comment]
我已经抛弃paopao了，目前是ikuai+mac mini m2 + surge ，paodns连接数过高会经常重播很烦。

TwoBall ***@***.***> 于2024年4月29日周一 12:59写道：
